From d406bde3e914cae2d794bbaab66aad6b7038b30e Mon Sep 17 00:00:00 2001
From: Mark Wielaard <mark@klomp.org>
Date: Fri, 17 Mar 2017 10:51:36 +0100
Subject: [PATCH] build/files.c (processPackageFiles): Don't call
 generateBuildIDs for noarch.

We don't want to do build-id processing for noarch packages. It might be
that noarch packages do contain architecture depended files, but those are
already handled by processBinaryFiles.

Signed-off-by: Mark Wielaard <mark@klomp.org>
---
 build/files.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/build/files.c b/build/files.c
index 6021643..35b2dd0 100644
--- a/build/files.c
+++ b/build/files.c
@@ -2384,10 +2384,14 @@ static rpmRC processPackageFiles(rpmSpec spec, rpmBuildPkgFlags pkgFlags,
 	goto exit;
 
 #if HAVE_LIBDW
-    if (generateBuildIDs (&fl) != 0) {
-	rpmlog(RPMLOG_ERR, _("Generating build-id links failed\n"));
-	fl.processingFailed = 1;
-	goto exit;
+    /* Check build-ids and add build-ids links for files to package list. */
+    const char *arch = headerGetString(pkg->header, RPMTAG_ARCH);
+    if (!rstreq(arch, "noarch")) {
+	if (generateBuildIDs (&fl) != 0) {
+	    rpmlog(RPMLOG_ERR, _("Generating build-id links failed\n"));
+	    fl.processingFailed = 1;
+	    goto exit;
+	}
     }
 #endif
 
-- 
1.8.3.1

